handler/handler_test.go

```diff
@@ -1,56 +1,107 @@
-	DataType:        "testDataType",
-	JobID:           "",
-	SearchIndex:     "ONS",
-	CDID:            "",
-	DatasetID:       "",
-	Keywords:        []string{"testkeyword1"},
-	MetaDescription: "",
-	Summary:         "",
-	ReleaseDate:     "",
-	Title:           "testTitle1",
-	TraceID:         "testTraceID",
-}
+var (
+	testContext = context.Background()
+
+	expectedEvent1 = &models.SearchDataImportModel{
+		DataType:        "testDataType1",
+		JobID:           "",
+		SearchIndex:     "ONS",
+		CDID:            "",
+		DatasetID:       "",
+		Keywords:        []string{"testkeyword1"},
+		MetaDescription: "",
+		Summary:         "",
+		ReleaseDate:     "",
+		Title:           "testTitle1",
+		TraceID:         "testTraceID1",
+	}
+
+	expectedEvent2 = &models.SearchDataImportModel{
+		DataType:        "testDataType2",
+		JobID:           "",
+		SearchIndex:     "ONS",
+		CDID:            "",
+		DatasetID:       "",
+		Keywords:        []string{"testkeyword2"},
+		MetaDescription: "",
+		Summary:         "",
+		ReleaseDate:     "",
+		Title:           "testTitle2",
+		TraceID:         "testTraceID2",
+	}
+
+	testEvents = []*models.SearchDataImportModel{
+		expectedEvent1,
+		expectedEvent2,
+	}
+
+	testSigner *awsauth.Signer
+
+	testCfg, _ = config.Get()
 
-var expectedEvent2 = &models.SearchDataImportModel{
-	DataType:        "testDataType2",
-	JobID:           "",
-	SearchIndex:     "ONS",
-	CDID:            "",
-	DatasetID:       "",
-	Keywords:        []string{"testkeyword2"},
-	MetaDescription: "",
-	Summary:         "",
-	ReleaseDate:     "",
-	Title:           "testTitle2",
-	TraceID:         "testTraceID2",
+	emptyListOfPathsWithNoRetries = func() []string {
+		return []string{}
+	}
+
+	setListOfPathsWithNoRetries = func(listOfPaths []string) {
+		return
+	}
+)
+
+func successESResponse() *http.Response {
+
+	return &http.Response{
+		StatusCode: 201,
+		Body:       ioutil.NopCloser(bytes.NewBufferString(`Created`)),
+		Header:     make(http.Header),
+	}
 }
 
-var expectedEvents = []*models.SearchDataImportModel{
-	expectedEvent1,
-	expectedEvent2,
+func clientMock(doFunc func(ctx context.Context, request *http.Request) (*http.Response, error)) *dphttp.ClienterMock {
+	return &dphttp.ClienterMock{
+		DoFunc:                    doFunc,
+		GetPathsWithNoRetriesFunc: emptyListOfPathsWithNoRetries,
+		SetPathsWithNoRetriesFunc: setListOfPathsWithNoRetries,
+	}
 }
 
-func TestPublishedContentExtractedHandler_Handle(t *testing.T) {
+func TestDataImporterHandle(t *testing.T) {
+
+	Convey("Given a handler configured with sucessful es updates", t, func() {
+		esDestURL := "locahost:9999"
+
+		//ES Client initialisation : this needs to be mock
+		doFuncWithInValidResponse := func(ctx context.Context, req *http.Request) (*http.Response, error) {
+			return successESResponse(), nil
+		}
+		httpCli := clientMock(doFuncWithInValidResponse)
+		esTestclient := dpelasticsearch.NewClientWithHTTPClientAndAwsSigner(esDestURL, testSigner, true, httpCli)
```

Comment:
So no need for `testSigner`, you can change this to:

```suggestion
		esTestclient := dpelasticsearch.NewClientWithHTTPClientAndAwsSigner(esDestURL, nil, false, httpCli)
```

and it will pass - the bool value tells the client to update http round tripper to add aws credentials via the signer using `testSigner`.

Reply:
The same logic can extend to other tests